src/commons/sideContent/SideContentContestVoting.tsx

```diff
@@ -17,79 +17,183 @@ type StateProps = {
-              canSave={canSave}
-              entryNumber={index + 1}
-              key={contestEntry.submission_id}
-              handleContestEntryClick={handleContestEntryClick}
-              handleVotingSubmissionChange={handleVotingSubmissionChange}
-              contestEntry={contestEntry}
-              minScore={minScore}
-              maxScore={maxScore}
-            />
-          ))
+      <div className="tier-list">
+        {tierBoard}
+        {sortedContestEntries.length > 0 ? (
+          <div className="tier" id="bank">
+            <div
+              className="item-container"
+              id="items"
+              onDragOver={handleDragOver}
+              onDragLeave={handleDragLeave}
+              onDragEnter={handleDragEnter}
+              onDrop={handleDrop}
+            >
+              {sortedContestEntries.map((contestEntry: ContestEntry, index) => (
+                <div
+                  className={classNames('item', { wrong: !isValid })}
+                  draggable={canSave}
+                  key={`item-${index + 1}`}
+                  id={`item-${index + 1}`}
+                  onDragStart={handleDragStart}
+                  onDragEnd={handleDragEnd(contestEntry)}
+                >
+                  <Card
+                    className={Classes.INTERACTIVE}
+                    elevation={Elevation.ONE}
+                    onClick={() =>
+                      handleContestEntryClick(
+                        contestEntry.submission_id,
+                        contestEntry.answer.code ?? ''
+                      )
+                    }
+                  >
+                    <Pre className="contestentry-entryid">{index + 1}</Pre>
+                  </Card>
+                </div>
+              ))}
+            </div>
+          </div>
         ) : (
           <div className="noResults">There are no eligible entries for voting found.</div>
         )}
       </div>
     ),
-    // determines when to re-render
     [
       isValid,
       canSave,
-      contestEntries,
+      sortedContestEntries,
       handleContestEntryClick,
-      handleVotingSubmissionChange,
-      maxScore,
-      minScore
+      handleDragEnd,
+      handleDragEnter,
+      handleDragLeave,
+      handleDragOver,
+      handleDrop,
+      tierBoard
     ]
   );
 
+  useEffect(() => {
+    sortedContestEntries.forEach((entry, index) => {
+      if (entry.score !== null) {
+        const tierIndex = TIERS.findIndex(tier => tier.score === entry.score);
+        if (tierIndex !== -1) {
+          const tierElement = document.querySelector(
+            `#tier-${TIERS[tierIndex].name.toLowerCase()} .item-container`
+          );
+          const bankElement = document.querySelector(`#bank .item-container #item-${index + 1}`);
+          if (tierElement && bankElement) {
+            tierElement.appendChild(bankElement);
+          }
+        }
+      }
+    });
+  }, [contestEntryCards, sortedContestEntries]);
+
```

Comment:
I'm not entirely sure what this `useEffect` is doing. Do you mind explaining?

Reply:
Hi, this useEffect has 2 main purposes:
1. Enabling the contestVoting to "save" the tiers of the contest entries. Previously, the cards would automatically be assigned to the bank upon re-opening the assignment, even if the user already had a saved submission. Thus, we came up with a method to automatically map entries back to their previously saved tier upon reopening. We wanted to implement it in contestVotingContainer (as it is the pre-render score check) but this was not possible due to the fact that a card (the bankElement) needed to be rendered _first_ before it could be automatically assigned to the tier it was previously saved in by the user. Hence, we implemented it to take effect upon a change to contestEntryCards (i.e. the initial build of contestEntryCards would trigger it), which would be after the creation of the cards but before the render of SideContentContestVoting, i.e. at just the right time so that on first opening, the viewer would see the correctly saved submission that they had inputted earlier.
2. This useEffect also opened the door to migrating legacy submissions (using the old score-based system) to the new tier system, and I am happy to say that we have successfully implemented it in the latest upvoted file! Now, all past submissions will show a tier based on the score the user assigned them in the past, using our new function mapTiers which maps 10 -> S; 8, 9 -> A etc, and the useEffect function implemented. This means that if the user reopens a past submission from say AY 22/23 Sem 1, they will be able to see their past vote with the new Tierlist design, and all the entries saved in a tier instead of a numerical score. This is also useful for keeping data from contests using the old system.